core-web/libs/sdk/client/src/lib/client/content/builders/collection/collection.ts

```diff
@@ -0,0 +1,334 @@
+     *
+     * @param {string} query A Lucene Query String
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    query(query: string): this;
+
+    /**
+     * Takes a function that recieves a QueryBuilder to buid a query for content filtering.
+     * @example
+     *```javascript
+     * // This will filter the content by title equals 'Hello World' or 'Hello World 2'
+     * client.content.getCollection("Activity").query((queryBuilder) =>
+     *     queryBuilder.field('title').equals('Hello World').or().equals('Hello World 2')
+     * );
+     *```
+     * @param {BuildQuery} buildQuery A function that receives a QueryBuilder instance and returns a valid query
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    query(buildQuery: BuildQuery): this;
+    query(buildQuery: BuildQuery | string): this {
+        if (typeof buildQuery === 'string') {
+            this.#rawQuery = buildQuery;
+
+            return this;
+        }
+
+        if (typeof buildQuery !== 'function') {
+            throw new Error('Parameter for query method should be a function or a string');
+        }
+
+        const builtQuery = buildQuery(new QueryBuilder());
+
+        // This can be use in Javascript so we cannot rely on the type checking
+        if (builtQuery instanceof Equals) {
+            this.#query = builtQuery.raw(this.currentQuery.build());
+        } else {
+            throw new Error('Provided query is not valid.');
+        }
+
+        return this;
+    }
+
+    /**
+     * Takes a boolean that represents if the content to query should be on draft or not
+     *
+     * Note: The default value is false to fetch content that is not on draft
+     *
+     * @param {boolean} draft A boolean that represents the state of the content
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    draft(draft: boolean): this {
+        this.#query = this.currentQuery.field('live').equals((!draft).toString());
+
+        return this;
+    }
+
+    /**
+     * Takes a string that represents a variant ID of content created with the A/B Testing feature
+     *
+     * Note: variantId defaults to "DEFAULT" to fetch content that is not part of an A/B test
+     *
+     * @param {string} variantId A string that represents a variant ID
+     * @return {CollectionBuilder} CollectionBuilder - A CollectionBuilder instance
+     * @memberof CollectionBuilder
+     */
+    variant(variantId: string): this {
+        this.#query = this.currentQuery.field('variant').equals(variantId);
+
+        return this;
+    }
+
+    /**
+     * Takes a number that represents the depth of the relationships of a content
+     *
+     * Note: The depth is set to 0 by default and the max supported value is 3
```

Comment:
I think there is a jsdoc better suitable for this.

Reply:
There is not JSDocs for notes but I removed the `Note:` it looks more like a description now